app/org/sagebionetworks/bridge/dynamodb/DynamoCompoundActivityDefinitionDao.java

```diff
@@ -0,0 +1,133 @@
+
+/** DynamoDB implementation of CompoundActivityDefinitionDao. */
+@Component
+public class DynamoCompoundActivityDefinitionDao implements CompoundActivityDefinitionDao {
+    private DynamoDBMapper mapper;
+
+    /** DDB mapper, configured by Spring. */
+    @Resource(name = "compoundActivityDefinitionDdbMapper")
+    public final void setDdbMapper(DynamoDBMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public CompoundActivityDefinition createCompoundActivityDefinition(StudyIdentifier studyId,
+            CompoundActivityDefinition compoundActivityDefinition) {
+        // Currently, all CompoundActivityDefinitions are DynamoCompoundActivityDefinitions.
+        DynamoCompoundActivityDefinition ddbDef = (DynamoCompoundActivityDefinition) compoundActivityDefinition;
+
+        // Set study to prevent people from creating defs in other studies.
+        ddbDef.setStudyId(studyId.getIdentifier());
+
+        // Clear the version. This allows people to copy-paste defs.
+        ddbDef.setVersion(null);
+
+        // Call DDB to create.
+        try {
+            mapper.save(ddbDef);
+        } catch (ConditionalCheckFailedException ex) {
+            throw new ConcurrentModificationException(ddbDef);
+        }
+
+        return ddbDef;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void deleteCompoundActivityDefinition(StudyIdentifier studyId, String taskId) {
+        // For whatever reason, DynamoDBMapper requires you to load the object before you delete it. It seems like you
+        // can't use Delete Expressions to make this a single atomic request.
+        DynamoCompoundActivityDefinition loadedDef = (DynamoCompoundActivityDefinition) getCompoundActivityDefinition(
+                studyId, taskId);
+
+        // Call DDB to delete.
+        try {
+            mapper.delete(loadedDef);
+        } catch(ConditionalCheckFailedException e) {
+            // in case of race condition
+            throw new EntityNotFoundException(CompoundActivityDefinition.class);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public List<CompoundActivityDefinition> getAllCompoundActivityDefinitionsInStudy(StudyIdentifier studyId) {
+        // query expression
+        DynamoCompoundActivityDefinition ddbHashKey = new DynamoCompoundActivityDefinition();
+        ddbHashKey.setStudyId(studyId.getIdentifier());
+        DynamoDBQueryExpression<DynamoCompoundActivityDefinition> ddbQueryExpr =
+                new DynamoDBQueryExpression<DynamoCompoundActivityDefinition>().withHashKeyValues(ddbHashKey);
+
+        // execute query
+        List<DynamoCompoundActivityDefinition> defList = mapper.query(DynamoCompoundActivityDefinition.class,
+                ddbQueryExpr);
+
+        // because of generics wonkiness, we need to convert this to a list of parent class CompoundActivityDefinition
+        return defList.stream().collect(Collectors.toList());
```

Comment:
I found that ImmutableList.copyOf(defList) also works, I can't find a way to do this that doesn't iterate over the results